semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,477 @@
+
+        surf = Surface(
+            nx=nx, ny=ny, xinc=xinc, yinc=yinc, xstart=xstart, ystart=ystart, angle=0
+        )
+
+        irap_x = np.empty(nx * ny)
+        irap_y = np.array(irap_x)
+
+        for i in range(len(surf)):
+            irap_x[i], irap_y[i] = surf.getXY(i)
+
+        # Interpolate vel grid to irap grid, should be the same apart from ordering
+        z = np.nan_to_num(z)
+        ip = CloughTocher2DInterpolator((x, y), z, fill_value=0)
+        irap_z = ip(irap_x, irap_y)
+
+        for i in range(len(surf)):
+            surf[i] = irap_z[i]
+
+        return surf
+
+    def add_survey(self, name, date):
+        """The add_survey() method will register a survey at a specific date.
+
+        The name argument should be a unique string, this will later
+        be used when evaluating the elastic strain in the
+        overburden. The date should be python datetime.date()
+        instance, this date should be present as a report step in the
+        restart file - otherwise an exception will be raised.
+        """
+        restart_view = self._rst_file.restartView(sim_time=date)
+        self.subsidence.add_survey_PRESSURE(name, restart_view)
+
+        self._restart_views[name] = restart_view
+
+        return restart_view
+
+    @staticmethod
+    def _divide_negative_shift(ts, div_val=5.0):
+        for i in range(len(ts)):
+            if ts[i] < 0:
+                ts[i] /= div_val
+
+    def geertsma_ts_rporv(self, vintage_pairs):
+        """
+        Calculates TS without using velocity. Fast.
+        Velocity is only used to get the surface on the velocity grid.
+        Uses change in porevolume from Eclipse (RPORV in .UNRST) as input to
+        Geertsma model.
+
+        :param vintage_pairs:
+        """
+        return self._geertsma_ts_custom(
+            vintage_pairs, self.subsidence.eval_geertsma_rporv, "TS_RPORV"
+        )
+
+    def geertsma_ts_simple(self, vintage_pairs):
+        """
+        Calculates TS without using velocity. Fast.
+        Velocity is only used to get the surface on the velocity grid.
+
+        :param vintage_pairs:
+        """
+        return self._geertsma_ts_custom(
+            vintage_pairs, self.subsidence.evalGeertsma, "TS_SIMPLE"
+        )
+
+    def _geertsma_ts_custom(self, vintage_pairs, subsidence_func, method_name):
+        """
+        Calculates TS without using velocity. Fast.
+
+        :param vintage_pairs:
+        :param subsidence_func: specify subsidence method to be used
+        :param method_name: string represeting the subsudence func name
+        """
+
+        if len(vintage_pairs) < 1:
+            return 0, []
+
+        vintages = self._vintages_name_date(vintage_pairs)
+        surface = self._surface
+        points_to_calculate = self._get_non_nan_points()
+        surf_displacement = np.zeros((len(vintages.name), len(surface)))
+        ts_surfaces = []
+        for iv, (vn, vd) in enumerate(zip(vintages.name, vintages.date)):
```

Comment:
Can you rename `iv`, `vn` and `vd`, better to just call them `vintages_name` and `vintages_date`, or just `name` and `date`?

Reply:
I thought about the following: `idx_vtg`, `name_vtg`, `date_vtg`